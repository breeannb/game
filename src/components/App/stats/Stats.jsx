import React from 'react';
import PropTypes from 'prop-types';

const Stats = ({ hp, attack, defense, magic, strength, items  }) => {
  return (
    <section>
      <p>HP: {hp}</p>
      <p>Attack: {attack}</p>
      <p>Defense: {defense}</p>
      <p>Magic: {magic}</p>
      <p>Strength: {strength}</p>
      <p>Items: {items}</p>
    </section>
  );
};

Stats.propTypes = {
  hp: PropTypes.number.isRequired, 
  attack: PropTypes.number.isRequired, 
  defense: PropTypes.number.isRequired, 
  magic: PropTypes.number.isRequired,
  strength: PropTypes.number.isRequired,
  items: PropTypes.number.isRequired,

};

export default Stats;

