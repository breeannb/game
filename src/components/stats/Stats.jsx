import React from 'react';
import PropTypes from 'prop-types';

const Stats = ({ speed, might, sanity, knowledge  }) => {
  return (
    <section>
      <p>Speed: {speed}</p>
      <p>Might: {might}</p>
      <p>Sanity: {sanity}</p>
      <p>Knowledge: {knowledge}</p>
    </section>
  );
};

Stats.propTypes = {
  speed: PropTypes.number.isRequired, 
  might: PropTypes.number.isRequired, 
  sanity: PropTypes.number.isRequired, 
  knowledge: PropTypes.number.isRequired,
};

export default Stats;

