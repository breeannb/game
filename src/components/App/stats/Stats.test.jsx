import React from 'react';
import { shallow } from 'enzyme';
import Stats from './Stats';

describe('Stats component', () => {
  it('renders Stats', () => {
    const wrapper = shallow(<Stats />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
