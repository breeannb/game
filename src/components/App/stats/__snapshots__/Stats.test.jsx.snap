// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats component renders Stats 1`] = `
<section>
  <p>
    HP: 
  </p>
  <p>
    Attack: 
  </p>
  <p>
    Defense: 
  </p>
  <p>
    Magic: 
  </p>
  <p>
    Strength: 
  </p>
  <p>
    Items: 
  </p>
</section>
`;
